packages/commerce-sdk-react/src/hooks/ShopperCustomers/cache.ts

```diff
@@ -17,81 +19,231 @@ import {
 import {and, pathStartsWith} from '../utils'
 
 type Client = ApiClients['shopperCustomers']
+type Customer = ShopperCustomersTypes.Customer
+type CustomerProductList = ShopperCustomersTypes.CustomerProductList
+type CustomerProductListResult = ShopperCustomersTypes.CustomerProductListResult
 
 const noop = () => ({})
 
-/** Invalidates the customer endpoint, but not derivative endpoints. */
-const invalidateCustomer = (parameters: Tail<QueryKeys['getCustomer']>): CacheUpdate => ({
-    invalidate: [{queryKey: getCustomer.queryKey(parameters)}]
-})
+// ReactQuery type def for the updater function uses type `unknown` for the first param,
+// this makes it hard to define the `createUpdateFunction` below, so we created this type.
+type Updater<T> = (data: T | undefined) => T | undefined
```

Comment:
This is the type used by React Query:
```ts
export declare type DataUpdateFunction<TInput, TOutput> = (input: TInput) => TOutput;
export declare type Updater<TInput, TOutput> = TOutput | DataUpdateFunction<TInput, TOutput>;
```
I don't know what the issue was you were encountering, but it seems like it was TS failing at inferring the type, rather than an issue with React Query's type def.

(Not saying you need to use React Query's definition, though. It barely amounts to anything.)

Reply:
I gave it one more shot using those react query types.. but I couldn't crack it.